packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -174,19 +174,28 @@ export const addEdge = (
 	targetInputPort.status = WorkflowPortStatus.CONNECTED;
 };
 
-export const removeEdge = (wf: Workflow, id: string) => {
+export const removeEdge = (wf: Workflow, id: string, registry: WorkflowRegistry) => {
 	const edgeToRemove = wf.edges.find((d) => d.id === id);
 	if (!edgeToRemove) return;
 
 	// Remove the data reference at the targetPort
 	const targetNode = wf.nodes.find((d) => d.id === edgeToRemove.target);
 	if (!targetNode) return;
-	const targetPort = targetNode.inputs.find((d) => d.id === edgeToRemove.targetPortId);
+	const targetPortIndex = targetNode.inputs.findIndex((d) => d.id === edgeToRemove.targetPortId);
+	const targetPort = targetNode.inputs[targetPortIndex];
+
 	if (!targetPort) return;
 	targetPort.value = null;
 	targetPort.status = WorkflowPortStatus.NOT_CONNECTED;
 	delete targetPort.label;
 
+	// Resets the type to the initial input type if it was a union type
+	const initialInput = registry.getOperation(targetNode.operationType)?.inputs?.[targetPortIndex];
+	if (initialInput?.type.includes('|')) {
+		targetPort.type = initialInput.type;
+		targetPort.label = initialInput.label;
+	}
+
```

Comment:
I don't think this is a good use of the registry. 

Even though from a technical sense it works it is not a very cohesive idea - if we want to retain or to revert type information, it needs to be built into the operator definition themselves, rather than going through a proxy and relying on indices to match up. In the longer term this can create unwanted baggage.